### PR TITLE
cxx-qt: Add CxxQtType trait

### PR DESCRIPTION
**Pull request by LeonMatthesKDAB**
This trait is automatically implemented on any `qobject::T`. It currently exposes the inner Rust type for a given QObject.

First step of implementing #550

This also requires us to split our macros into a cxx-qt-macro crate, as otherwise only exporting proc-macros is allowed.
We could now potentially move CxxQtThread into the cxx-qt crate now to remove the dependency on cxx-qt-lib.

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -33,6 +33,7 @@ version = "0.5.3"
 # we publish, otherwise crates.io complains as it doesn't know the version.
 [workspace.dependencies]
 cxx-qt = { path = "crates/cxx-qt" }
+cxx-qt-macro = { path = "crates/cxx-qt-macro" }
```
Comment (ahayzen-kdab) [inline Cargo.toml:36]: can you split this split into a separate pull request. As the trait stuff we might want to see things develop a little bit further to check things work before merging. And it eases review if things are small

Comment (LeonMatthesKDAB) [reply Cargo.toml:36]: Done

File: crates/cxx-qt-gen/src/writer/rust/mod.rs [hunk 2]
```diff
@@ -41,9 +41,9 @@ fn cxx_bridge_common_blocks(qobject: &GeneratedRustQObject) -> Vec<TokenStream>
                 /// <https://github.com/dtolnay/cxx/issues/683>
                 type #cxx_qt_thread_ident;
 
-                /// Retrieve an immutable reference to the Rust struct backing this C++ object
                 #[cxx_name = "unsafeRust"]
-                fn rust(self: &#cpp_struct_ident) -> &#rust_struct_ident;
+                #[doc(hidden)]
+                fn ffi_rust(self: &#cpp_struct_ident) -> &#rust_struct_ident;
```
Comment (ahayzen-kdab) [inline crates/cxx-qt-gen/src/writer/rust/mod.rs:46]: do we use `ffi_` as our common "hidden" method naming ? As we have the same problem with `qt_thread()` ? or should we use like `cxx_qt_` or `cxx_qt_ffi_` ?

Namespacing to our things to at least `cxx_qt` would avoid any collisions accidentally

Comment (LeonMatthesKDAB) [reply crates/cxx-qt-gen/src/writer/rust/mod.rs:46]: We could do `ccx_qt_ffi` :thinking: 
Let me check if that also fixes our "similarity" issues.

Comment (LeonMatthesKDAB) [reply crates/cxx-qt-gen/src/writer/rust/mod.rs:46]: image
At least for `rust()` and `rust_mut()`, you no longer get suggested the `ffi_rust` stuff.

@ahayzen-kdab Can you check that `cxx_qt_ffi` also works for `qt_thread`?
As that may have more overlap with the longer method name.

Comment (ahayzen-kdab) [reply crates/cxx-qt-gen/src/writer/rust/mod.rs]: Yup it makes it different enough too

```
build] error[E0599]: no method named `qt_thread` found for struct `Pin<&mut threading::ffi::ThreadingWebsiteQt>` in the current scope
[build]   --> examples/qml_features/rust/src/threading.rs:90:34
[build]    |
[build] 90 |             let qt_thread = self.qt_thread();
[build]    |                                  ^^^^^^^^^ method not found in `Pin<&mut ThreadingWebsiteQt>`
[build]    |
[build]    = help: items from traits can only be used if the trait is in scope
[build] help: the following trait is implemented but not in scope; perhaps add a `use` for it:
[build]    |
[build] 11 | use cxx_qt::Threading;
[build]    |
```

**Comment by ahayzen-kdab** [comment]
Just needs rebasing, then looks good :-)
